linked_list/linkedList.js

```diff
@@ -0,0 +1,272 @@
+			for (let i = 0; i < this._size; i++) {
+				if (i === index) {
+					head.data = input;
+				}
+				head = head.next;
+			}
+		}
+	}
+
+	// Clones and returns a newList.
+	clone () {
+		let newList = new LinkedList();
+		let temp = this.head;
+		while (temp.next) {
+			newList.addLast(temp.data);
+			temp = temp.next;
+		}
+		newList.addLast(temp.data);
+		return newList;
+	}
+
+	// Reverses the existing Linked List.
+	reverse () {
+		let prev = null;
+		let current = this.head;
+		while (current) {
+			let next = current.next;
+			current.next = prev;
+			prev = current;
+			current = next;
+		}
+		this.head = prev;
+	}
+
+	// Checks if Linked List is empty.
+	isEmpty () {
+		return (this._size === 0);
+	}
+
+	// Clears the Linked List.
+	clear () {
+		this.head = null;
+		this._size = 0;
+	}
+
+	// Returns the Linked List in List form.
+	showList () {
+		let list = [];
+		let head = this.head;
+		while (head) {
+			list.push(head.data);
+			head = head.next;
+		}
+		return list;
+	}
+}
+
+function main () {
+	let linkedList = new LinkedList();
+
+	linkedList.add(5, 0);
+	console.log(linkedList.showList());
+	console.log('Size : ' + linkedList._size);
+
+	linkedList.addFirst(25);
+	linkedList.addFirst(251);
+	console.log(linkedList.showList());
+	console.log('Size : ' + linkedList._size);
+
+	linkedList.addFirst(5);
+	console.log(linkedList.showList());
+	console.log('Size : ' + linkedList._size);
+
+	try {
+		console.log('Fist Occurance at index: ' + linkedList.findFirst(5));
+	} catch (err) {
+		console.log('Value not found');
+	}
+
+	try {
+		console.log('Last Occurance at index: ' + linkedList.findLast(5));
+	} catch (err) {
+		console.log('Value not found');
+	}
+
+	try {
+		linkedList.setData(12325, 1);
+		console.log(linkedList.showList());
+		console.log('Size : ' + linkedList._size);
+	} catch (err) {
+		console.log('Index out of bound');
+	}
+
+	try {
+		linkedList.removeByValue(12325);
+		console.log(linkedList.showList());
+		console.log('Size : ' + linkedList._size);
+	} catch (err) {
+		console.log('Value not found');
+	}
+
+	linkedList.addLast(125);
+	linkedList.addLast(11);
+	console.log(linkedList.showList());
+	console.log('Size : ' + linkedList._size);
+
+	try {
+		console.log('Element found at index: ' + linkedList.search(1125));
+	} catch (err) {
+		console.log('Value not found');
+	}
+
+	console.log('Cloned : ', linkedList.clone().showList());
+
+	try {
+		linkedList.remove(6);
+		console.log(linkedList.showList());
+		console.log('Size : ' + linkedList._size);
+	} catch (err) {
+		console.log('Index Out of Bound');
+	}
+
+	try {
+		linkedList.remove(2);
+	} catch (err) {
+		console.log('Index Out of Bound');
+	}
+	console.log(linkedList.showList());
+	console.log('Size : ' + linkedList._size);
+
+	linkedList.removeFront();
+	console.log(linkedList.showList());
+	console.log('Size : ' + linkedList._size);
+
+	linkedList.removeLast();
+	console.log(linkedList.showList());
+	console.log('Size : ' + linkedList._size);
```

Comment:
In JavaScript, we usually have a `length` property. You should make this `length` too. 
https://www.w3schools.com/jsref/jsref_length_array.asp